### DISCUSSION TITLE
What does "Head over to the Home Assistant configuration and set up the integration there" mean?

### DISCUSSION THREAD

**Discussion by steerage250** [siku2/hass-weatherlink#35] [General]
It is not clear where I go to setup the integration?

**Comment by siku2** [comment]
After you've installed the custom integration and restarted Home Assistant, you can set up the integration from the default settings menu, the same way you would set up a normal integration.

Feel free to contribute better instructions if you ever figure it out.

**Comment by ByteTheHand** [comment]
I just went through the full setup tonight.  I setup my WeatherLinkLive, and then two AirLinks and everything was really super easy after I figured out getting HACS installed.

If I get some time, would you like me to take a shot at an Install Document?  While I thought it was easy once I got the library installed, I could add some documentation around that since some people interested in doing this might be new to pulling stuff from GitHub.

**Comment by steerage250** [comment]
An install document would be helpful.  However, after a reboot, my existing weatherlink integration (via configuration.yaml) started working again after I rebooted after installing the Weatherlink integration via HACS.  I'm a bit scared to start fiddling in case I'm left with nothing.

Really, I think my main problem is I can't find where to go to configure the integration.  I HACS, it is showing that it needs an update, so it is installed - I just don't know where to go to set it up (and that's probably all I need)

thanks

**Comment by ByteTheHand** [comment]
Once you have gone into HACS and added the Github link you should see this in HACS

![image](https://user-images.githubusercontent.com/45699811/233435573-6e843966-2725-47b1-993f-c680e899b6bc.png)

Then go into Settings > Devices & Services to the Integrations tab.  Click the "+ Add Integration" button and search for WeatherLInk.

![image](https://user-images.githubusercontent.com/45699811/233436240-49baffb2-115e-4118-abc4-96fdf3808321.png)

Clicking on WeatherLink will display the following to set up your host:

![image](https://user-images.githubusercontent.com/45699811/233436420-79530a0a-c0e0-4729-855e-95db5c798e50.png)

I have my WeatherLinkLive and AirLinks all on reserved IP addresses, so I just put in the ip address like "192.168.0.##".  You should be able to put the name of each in there too, but I think the reserved IP route is best.

You will need to go through that for all of your devices.  I had to set it up three times to get all of mine set.  Once that is done, the integration will look like this:

![image](https://user-images.githubusercontent.com/45699811/233437090-f42b5964-7aad-4a03-8580-ac2a1696b949.png)

Click on one of the devices and you will see:

![image](https://user-images.githubusercontent.com/45699811/233437270-0c52b395-a0c0-49fd-9c72-c5519dbdea22.png)

Then click on "Configure" and the brings up a normal configuration panel:

![image](https://user-images.githubusercontent.com/45699811/233437511-2b88c85b-a491-431f-8299-c9bc14f332c6.png)

From there you can rename the device and add it to your dashboard like any other integration.  I just got it installed last night and so did a quick dashboard.  Still trying to figure out how to do and actually calculation to give a single AQI number based off of PM2.5.  While that isn't a true 100 AQI and it isn't the NowCast number, I'm only interested in a simple AQI to determine when to turn filter fans on and off and when to turn windows fans on and off based on internal and external air quality.

My first shot at a dashboard:

![image](https://user-images.githubusercontent.com/45699811/233438343-735eee46-6cda-4b33-b136-2c0a83483456.png)

**Reply by scotflieger** [reply]
I cannot get it to accept any host I try. I am trying to connect to my Vantage Vue 2 with wifilogger2 at 192.168.1.207. I don’t see any list like the example Aden #8. I have no AirLink devices. I would welcome some guidance.

**Reply by ByteTheHand** [reply]
Can you see your sensor data by going to your browser and entering http://192.168.1.207/v1/current_conditions ?  If not, then your logger might be changing IP addresses.  I have my WeatherLinkLive and three AirLinks all with reserved IP addresses on the router so I can address them directly.

If you can see it in your browser by entering that address, then the integration should be able to work with just the IP address (you don’t add the rest in the integration setup).  Having, or not having AirLink sensor shouldn’t be and issue, I just have to add more sensors during setup.  So I have the integration looking at 0.30, 0.31, 0.32, and 0.33 to cover all of my data.  You’ll just use the single address of …1.207.

Let me know if that makes sense and if it works.  Either way, I can try to help you setup or trouble shoot.

**Reply by ByteTheHand** [reply]
I’ll add that I’ve actually expanded now to four AirLink sensors and have played with another dashboard that I like for all of the AirLink Sensores.

<img width="1366" alt="image" src="https://github.com/siku2/hass-weatherlink/assets/45699811/19b62831-af84-46ec-a343-c3c2a340fde4">

**Reply by scotflieger** [reply]
When I use the link you suggested I get FileNotFound returned. I have fixed the IP address (192.168.1.207) at my router.

**Reply by ByteTheHand** [reply]
If your IP address is now reserved for your logger, the next question is the layout the same as what they are looking for.  Below is what my WeatherLinkLive reports. I would think that Davis would keep this part consistent since they have to log it all into the WeatherLink site, but check if your data layout follows this format:

<img width="1366" alt="image" src="https://github.com/siku2/hass-weatherlink/assets/45699811/3389ef71-f033-41e1-9443-a3408cc6cd07">

**Reply by scotflieger** [reply]
I think I have worked out why I can't read anything using the integration. I have a wired Vantage Pro 2 and am trying to read data from my new WifiLogger 2 data logger. I don't have a Weatherlink Live device which you refer to.

**Reply by ByteTheHand** [reply]
Let me look at the WiFiLogger 2.  I just figured that it wrote to WeatherLink.com and if it does it might still be addressable.  Let me go look at the Davis site after work today and I’ll see if I can figure out what it is doing.  I originally used the IP data logger, so I’ll see if it is somewhat similar.

**Reply by scotflieger** [reply]
Thank you. I have looked at accessing the WifiLogger 2 (WL2) directly but can only see the Real Time Data (see below). The WL2 is said to push data to WeatherLink.com (WL.COM). I have successfully accessed WL.COM via http to obtain a .json extract of all data. I am working on trying to pull this data into HA.

![IMG_0814](https://github.com/siku2/hass-weatherlink/assets/8685475/532d26db-b059-42bf-9194-0f92b9d202c6)

**Reply by ByteTheHand** [reply]
I found this on WXForum.  It says to check http://[WFL_IP]/wflexpj.json and that will. Update every three seconds if it is polled again.  That might work, but it would still have to be in the same layout as Davis outputs from the console.  Worth looking into maybe.

**Reply by scotflieger** [reply]
That works using **http://192.168.1.207/wflexpj.json**.

`{"stnname":"WiFiLogger","stnmod":16,"ver":3.88,"mac":"94:E6:86:48:7D:0C","apmac":"94:E6:86:48:7D:0D","ip":"192.168.1.207","ssid":"BT-JFCWQN","rssi":-52,"wifimod":0,"lastboot":1693547751,"uptime":437135,"wflver":"2.42","loctime":1693988340,"utctime":1693984740,"tzone":18,"units":62,"conlati":577,"conlongi":-30,"wfllati":57.6528625,"wfllongi":-2.9729307,"tempout":58.7,"humout":90,"tempin":72.8,"humin":57,"windspd":2.0,"winddir":136,"windavg2":2.7,"windavg10":4.1,"gust":9.0,"gustdir":134,"bar":30.099,"bartr":-20,"dew":56,"cdew":55.8,"chill":59,"heat":59,"thsw":null,"uv":null,"solar":null,"rainr":0.00000,"storm":0.00000,"rain15":0.00000,"rain1h":0.00000,"raind":0.00000,"rain24":0.00000,"rainmon":0.00787,"rainyear":15.55115,"etday":0.000,"etmon":0.00,"etyear":0.00,"xt":[null,null,null,null,null,null,null],"xlt":[null,null,null,null],"xst":[null,null,null,null],"xh":[null,null,null,null,null,null,null],"xsm":[null,null,null,null],"xlw":[null,null,null,0],"bat":4.72,"trbat":0,"foreico":6,"forrule":44,"sunrt":"6:23","sunst":"19:57","hlbar":[30.099,30.178,"8:18","0:42",30.216,29.819,30.796,28.707],"hlwind":[null,20.0,null,"2:50",28.0,null,60.0,null],"hltempin":[72.8,74.8,"6:36","0:01",76.4,70.2,79.0,62.0],"hlhumin":[56,58,"4:38","3:03",60,53,69,33],"hltempout":[56.7,58.7,"4:47","8:15",76.9,46.6,83.0,25.0],"hlhumout":[88,91,"0:46","7:03",93,51,95,28],"hldew":[54,56,"0:00","7:44",62,44,67,22],"hlchil":[53,null,"2:50",null,null,46,null,18],"hlheat":[null,59,null,"8:07",78,null,83,null],"hlthsw":[null,null,null,null,null,null,null,null],"hlsolar":[null,0,null,null,0,null,0,null],"hluv":[null,0.0,null,null,0.0,null,0.0,null],"hlrainr":[0.000,0.000,null,null,0.008,null,37.795,null],"hlxt0":[null,null,null,null,null,null,null,null],"hlxt1":[null,null,null,null,null,null,null,null],"hlxt2":[null,null,null,null,null,null,null,null],"hlxt3":[null,null,null,null,null,null,null,null],"hlxt4":[null,null,null,null,null,null,null,null],"hlxt5":[null,null,null,null,null,null,null,null],"hlxt6":[null,null,null,null,null,null,null,null],"hlxh0":[null,null,null,null,null,null,null,null],"hlxh1":[null,null,null,null,null,null,null,null],"hlxh2":[null,null,null,null,null,null,null,null],"hlxh3":[null,null,null,null,null,null,null,null],"hlxh4":[null,null,null,null,null,null,null,null],"hlxh5":[null,null,null,null,null,null,null,null],"hlxh6":[null,null,null,null,null,null,null,null],"hlxst0":[null,null,null,null,null,null,null,null],"hlxst1":[null,null,null,null,null,null,null,null],"hlxst2":[null,null,null,null,null,null,null,null],"hlxst3":[null,null,null,null,null,null,null,null],"hlxlt0":[null,null,null,null,null,null,null,null],"hlxlt1":[null,null,null,null,null,null,null,null],"hlxlt2":[null,null,null,null,null,null,null,null],"hlxlt3":[null,null,null,null,null,null,null,null],"hlxsm0":[null,null,"0:00",null,null,null,null,null],"hlxsm1":[null,null,"0:00",null,null,null,null,null],"hlxsm2":[null,null,"0:00",null,null,null,null,null],"hlxsm3":[null,null,"0:00",null,null,null,null,null],"hlxlw0":[null,null,"0:00",null,null,null,null,null],"hlxlw1":[null,null,"0:00",null,null,null,null,null],"hlxlw2":[null,null,"0:00",null,null,null,null,null],"hlxlw3":[null,null,"0:00",null,null,null,null,null]}`

**Reply by scotflieger** [reply]
Accessing Weatherlink.com using API V1 with **https://api.weatherlink.com/v1/NoaaExt.json?user=001D0AE0BA74&pass={user login password}&apiToken={v1 token}** extracts the following. It requires the user's Device ID (DID), WL.COM log in password and V1 token obtained from their WL.COM account. 

`{"credit":"Davis Instruments Corp.","credit_URL":"http:\/\/www.davisnet.com","disclaimer_url":"http:\/\/www.davisnet.com\/about\/terms.asp","copyright_url":"http:\/\/www.davisnet.com\/about\/terms.asp","privacy_policy_url":"http:\/\/www.davisnet.com\/about\/privacy.asp","image":{"url":"http:\/\/www.weatherlink.com\/images\/Logo_Davis_reflxblu.jpg","title":"Davis WeatherLink","link":"http:\/\/www.weatherlink.com"},"suggested_pickup":"15 minutes after the hour","suggested_pickup_period":"60","dewpoint_c":"15.6","dewpoint_f":"60.0","dewpoint_string":"60.0 F (15.6 C)","heat_index_c":"20.0","heat_index_f":"68.0","heat_index_string":"68.0 F (20.0 C)","location":"Buckie, Scotland, UK","latitude":"57.65287","longitude":"-2.97309","observation_time":"Last Updated on Sep 5 2023, 5:22 pm BST","observation_time_rfc822":"Tue, 5 Sep 2023 17:22:55 +0100","pressure_in":"30.175","pressure_mb":"1021.8","pressure_string":"1021.8 mb","relative_humidity":"76","station_id":"xxxxxxx","temp_c":"19.9","temp_f":"67.8","temperature_string":"67.8 F (19.9 C)","wind_degrees":"53","wind_dir":"Northeast","wind_kt":"1.7","wind_mph":"2.0","windchill_c":"20.0","windchill_f":"68.0","windchill_string":"68.0 F (20.0 C)","davis_current_observation":{"DID":"001D0AE0BA74","station_name":"Arradoul - WLP","observation_age":2,"dewpoint_day_high_f":"61","dewpoint_day_high_time":"3:12pm","dewpoint_day_low_f":"46","dewpoint_day_low_time":"5:16am","dewpoint_month_high_f":"62","dewpoint_month_low_f":"44","dewpoint_year_high_f":"67","dewpoint_year_low_f":"22","heat_index_day_high_f":"70","heat_index_day_high_time":"2:08pm","heat_index_month_high_f":"78","heat_index_year_high_f":"83","pressure_day_high_in":"30.216","pressure_day_high_time":"9:08am","pressure_day_low_in":"30.140","pressure_day_low_time":"12:33am","pressure_month_high_in":"30.216","pressure_month_low_in":"29.819","pressure_tendency_string":"Falling Slowly","pressure_year_high_in":"30.796","pressure_year_low_in":"28.707","rain_day_in":"0.0000","rain_month_in":"0.0100","rain_rate_day_high_in_per_hr":"0.0000","rain_rate_hour_high_in_per_hr":"0.0000","rain_rate_in_per_hr":"0.0000","rain_rate_month_high_in_per_hr":"0.0100","rain_rate_year_high_in_per_hr":"48.0000","rain_storm_in":"0.0000","rain_year_in":"19.7500","relative_humidity_day_high":"91","relative_humidity_day_high_time":"7:28am","relative_humidity_day_low":"66","relative_humidity_day_low_time":"10:46am","relative_humidity_month_high":"93","relative_humidity_month_low":"51","relative_humidity_year_high":"95","relative_humidity_year_low":"28","relative_humidity_in":"57","relative_humidity_in_day_high":"59","relative_humidity_in_day_high_time":"12:00am","relative_humidity_in_day_low":"55","relative_humidity_in_day_low_time":"1:54am","relative_humidity_in_month_high":"60","relative_humidity_in_month_low":"53","relative_humidity_in_year_high":"69","relative_humidity_in_year_low":"33","sunrise":"6:20am","sunset":"8:00pm","temp_day_high_f":"69.6","temp_day_high_time":"2:16pm","temp_day_low_f":"50.0","temp_day_low_time":"5:21am","temp_month_high_f":"76.9","temp_month_low_f":"46.6","temp_year_high_f":"83.0","temp_year_low_f":"25.0","temp_in_day_high_f":"75.6","temp_in_day_high_time":"5:17pm","temp_in_day_low_f":"72.7","temp_in_day_low_time":"8:19am","temp_in_f":"75.6","temp_in_month_high_f":"76.4","temp_in_month_low_f":"70.2","temp_in_year_high_f":"79.0","temp_in_year_low_f":"62.0","wind_day_high_mph":"14.0","wind_day_high_time":"3:40pm","wind_month_high_mph":"28.0","wind_ten_min_avg_mph":"2.0","wind_ten_min_gust_mph":"10.0","wind_year_high_mph":"60.0","windchill_day_low_f":"49","windchill_day_low_time":"5:26am","windchill_month_low_f":"46","windchill_year_low_f":"18"},"time_to_generate":"0.011172 seconds"}`

**Comment by steerage250** [comment]
Brilliant - thank you.  I hadn't realised it was a 2-step installation (ie HACS, then a "normal" Add Integration.

Unfortunately, after entering th IP address of my Weatherlink IP Datalogger device, it fails to connect.  I suspect it's interface might be too old?  Anyway, the other interface via the Weatherlink API is working, so not a problem.
